jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -778,26 +804,3 @@ def _fit_model_to_cube(self, add_data):
         self.hub.broadcast(snackbar_message)
 
         return fitted_model, output_cube
-
-    def _register_spectrum(self, event):
-        """
-        Add a spectrum to the data collection based on the currently displayed
-        parameters (these could be user input or fit values).
-        """
-        if self._warn_if_no_equation():
-            return
-        # Make sure the initialized models are updated with any user-specified
-        # parameters
-        self._update_initialized_parameters()
-
-        # Need to run the model fitter with run_fitter=False to get spectrum
-        if "spectrum" in event:
-            spectrum = event["spectrum"]
-        else:
-            model, spectrum = fit_model_to_spectrum(self._spectrum1d,
-                                                    self._initialized_models.values(),
-                                                    self.model_equation,
-                                                    window=self._window)
-
-        self.add_results.add_results_from_plugin(spectrum)
-        self._set_default_results_label()
```

Comment:
It seems that during past refactors this kind of got left as an unnecessary artifact that only gets called once internally (and so never even entered the else statement in 797).  This PR moves the remaining applicable logic to where it was called before.